### PR TITLE
[microsoft_dhcp] Fix incorrect log parsing for DHCP host.domain -> host.name

### LINKED ISSUE #4279: [microsoft_dhcp] Change host.domain to host.name to better align with ECS
Currently, the Microsoft DHCP integration has an ingest pipeline that is parsing the CSV and using the host.domain for the host name found in the DHCP logs. Not sure if this is an environment specific issue or not, but figured I would put some feelers out for others using this integration.

The fix is simply updating the pipeline for DHCP to use host.name instead of host.domain since the value contains: computer75.org.local which also contains the domain. If I can get a like or two, I will push forward and make the pull request.

The test document is here: https://github.com/elastic/integrations/blob/main/packages/microsoft_dhcp/data_stream/log/sample_event.json

The value is `host.domain : test-host` which reads to me as a host name not a domain.

### PR DESCRIPTION
**Pull request by nicpenning**
Closes #4279

## What does this PR do?
- Enhancement

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ processors:
         - _tmp_.time
         - message
         - host.ip
-        - host.domain
+        - host.name
```
Comment (efd6) [inline packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:13]: To help out people are using `host.domain`, maybe add a processor to get that. Something like:
```
  - grok:
      field: host.name
      if: 'ctx.host?.name != null && ctx.host.name.contains(".")'
      patterns:
      - "%{HOST}\\.%{GREEDYDATA:host.domain}"
      pattern_definitions:
        HOST: "[^.]+"
      ignore_failure: true
```

File: packages/microsoft_dhcp/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.6.1"
+  changes:
+    - description: Change host.domain to host.name to reflect the event data
+      type: enhancement
```
[the remark this answers is not in the document]
Comment (nicpenning) [reply packages/microsoft_dhcp/changelog.yml:5]: Okay, I can do that.

File: packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml (hunk 1 above)
Comment (nicpenning) [reply packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml:13]: I thought about that, but didn't want to over complicate this PR.

However, we might as well get domain out of the data. I will do that as well!

**Review by nicpenning — COMMENTED** [review]
Okay, I have made the appropriate changes. I don't have a way to test using .yml files for ingest pipelines, but the logic checks out when I updated my current pipeline in my test environment.

**Comment by efd6** [comment]
Please run `elastic-package build`.

**Comment by nicpenning** [comment]
Is that something I need to run or comment here?

**Comment by nicpenning** [comment]
I went ahead and built the package and at first had an issue with the meta.json because it had the old version. I updated that and was then able to build the microsoft_dhcp-1.7.0.zip package.

**Review by nicpenning — COMMENTED** [review]
Updated manifest.yml looks good.

**Review by nicpenning — COMMENTED** [review]
LGTM

**Comment by nicpenning** [comment]
I don't know why it failed on the test for the README.md. Please let me know how to proceed.

**Comment by efd6** [comment]
When I build the package locally, I see this diff.
```diff
diff --git a/packages/microsoft_dhcp/docs/README.md b/packages/microsoft_dhcp/docs/README.md
index dfdc14308..efd91c390 100644
--- a/packages/microsoft_dhcp/docs/README.md
+++ b/packages/microsoft_dhcp/docs/README.md
@@ -75,7 +75,7 @@ An example event for `log` looks as following:
     },
     "host": {
         "ip": "2a02:cf40:add:4002:91f2:a9b2:e09a:6fc6",
-        "name": "test-host"
+        "domain": "test-host"
     },
     "event": {
         "agent_id_status": "verified",
@@ -119,9 +119,9 @@ An example event for `log` looks as following:
 | event.timezone | This field should be populated when the event's timestamp does not include timezone information already (e.g. default Syslog timestamps). It's optional otherwise. Acceptable timezone formats are: a canonical ID (e.g. "Europe/Amsterdam"), abbreviated (e.g. "EST") or an HH:mm differential (e.g. "-05:00"). | keyword |
 | event.type | This is one of four ECS Categorization Fields, and indicates the third level in the ECS category hierarchy. `event.type` represents a categorization "sub-bucket" that, when used along with the `event.category` field values, enables filtering events down to a level appropriate for single visualization. This field is an array. This will allow proper categorization of some events that fall in multiple event types. | keyword |
 | host.domain | Name of the domain of which the host is a member. For example, on Windows this could be the host's Active Directory domain or NetBIOS domain name. For Linux this could be the domain of the host's LDAP provider. | keyword |
-| host.name | Name of the host. It can contain what hostname returns on Unix systems, the fully qualified domain name, or a name specified by the user. The sender decides which value to use. | keyword |
 | host.ip | Host ip addresses. | ip |
 | host.mac | Host MAC addresses. The notation format from RFC 7042 is suggested: Each octet (that is, 8-bit byte) is represented by two [uppercase] hexadecimal digits giving the value of the octet as an unsigned integer. Successive octets are separated by a hyphen. | keyword |
+| host.name | Name of the host. It can contain what `hostname` returns on Unix systems, the fully qualified domain name, or a name specified by the user. The sender decides which value to use. | keyword |
 | input.type |  | keyword |
 | log.file.path | Full path to the log file this event came from, including the file name. It should include the drive letter, when appropriate. If the event wasn't read from a log file, do not populate this field. | keyword |
 | log.offset |  | long |
```
I don't want to push this to your branch because you are sending the change from your master branch, and this complicates my repo, so can you make that change — this should be the result when you run `elastic-package build`.

**Comment by nicpenning** [comment]
Okay. Yeah, locally I don't get much out of the build:
```
$ elastic-package build
Build the package
README.md file rendered: /Users/.../integrations/packages/microsoft_dhcp/docs/README.md
2022/09/24 20:33:21  INFO License text found in "/Users/.../integrations/LICENSE.txt" will be included in package
Package built: /Users/.../integrations/build/packages/microsoft_dhcp-1.7.0.zip
Done
```

I can make those changes. That is a bad example anyways :/ test-host doesn't make sense for domain but rather host.name. In this example, host.domain wouldn't event exist as it would expect something like test-host.local to produce `host.domain:local` and `host.name:test-host.local`

Docs are important! I will see what I can do.

**Comment by nicpenning** [comment]
I tried your other command, but I am not running a stack so that could be part of my problem.

I updated the README.md to include the host.domain but updated the host.name example.

How does the test know what example to use? When I look at the test docs here:
https://github.com/nicpenning/integrations/blob/master/packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json

None of what is in the README.md aligns, so it must not be doing actually test with the data in conjunction with the README.md as I think you telling me.

Does my README.md update help?

**Comment by efd6** [comment]
Please apply this patch to your branch at 33f65b0. You will need to gunzip it first (GitHub does not allow patch files to be uploaded), and then `git apply 0001-microsoft_dhcp-bring-sample-expects-and-docs-up-to-d.patch` and then add and commit the changes.

[0001-microsoft_dhcp-bring-sample-expects-and-docs-up-to-d.patch.gz](https://github.com/elastic/integrations/files/9643407/0001-microsoft_dhcp-bring-sample-expects-and-docs-up-to-d.patch.gz)

**Comment by nicpenning** [comment]
By applying the patch locally on my system and pushing up, I apparently and inadvertently used the wrong credentials in my browser. I have an idea how that happened, but unsure how to resolve, yet.

**Comment by nicpenning** [comment]
Since I can't revert here on the web, we might have to figure out a way to remove these changes and re-apply unless you can override that message. 

I think this is a bug in the CLA since it was me that pushed, but since it doesn't include my email it fails to recognize me. Do you know if there is a workaround to allow First/Last name to show as having signed the CA?

**Review by nicpenning — COMMENTED** [review]
These all look good.

**Comment by efd6** [comment]
The CLA checker is not recognising the email address in the change. Can you `git reset --hard HEAD~1 && git push -f <your remote> master`. This will remove the commit and may make the CLA checker happy. Then redo the push as before but with the correct author details.

**Comment by efd6** [comment]
The alternative approach is to close this PR and make a new one using a named non-master branch. I can then help out with changes if needed.

**Comment by nicpenning** [comment]
I will take a run at your first suggestion!

**Comment by nicpenning** [comment]
Manually created new branch and added patch. :/ I have no idea how to use GitHub and publish with my email, it seems to always just generically send my First and Last name. Maybe it is a Git on Windows thing. Someday I will figure that out. Closing this PR in favor of the other branch I used. Thanks for all of your help. See you over on the new PR: https://github.com/elastic/integrations/pull/4311

**Pull request closed without merging**